File: 1_fibonacci_series/fibonacci_number.swift
// import Foundation
// import Cocoa

import Python
// %include "EnableIPythonDisplay.swift"
// IPythonDisplay.shell.enable_matplotlib("inline")
let plt = Python.import("matplotlib.pyplot")

let index = 90

if index >= 3 {
	var index_list = [1, 2]
	var fibo_list = [1, 1]
	var fibo1 = 1
	var fibo2 = 1
	var i = 3
	var fib_next = 0
	
	while i <= index {
		fib_next = fibo1 + fibo2
		index_list.append(i)
		fibo_list.append(fib_next)
		i += 1
		fibo1 = fibo2
		fibo2 = fib_next
	}
	// print(fibo_list)
	plt.plot(index_list, fibo_list)
	plt.show()
	
} else {
	print("at least first three of fibonacci number")
}

File: 1_fibonacci_series/fibonacci_number.py
import matplotlib.pyplot as plt

def fibonacci(Nth_fibonacci_number):
    x_axis = [i for i in range(1,Nth_fibonacci_number+1)]
    fibonacci_list = [1,1]
    for i in range(2, Nth_fibonacci_number): # min of N is 3 <=> min of index is 2
        new_fibonacci_number = fibonacci_list[i-1] + fibonacci_list[i-2]
        fibonacci_list.append(new_fibonacci_number)
    return(fibonacci_list, x_axis)


return_list, x_axis = fibonacci(8) # Nth_fibonacci_number>=2
print(return_list)
